Tests/Foundation/TestFileManager.swift

```diff
@@ -1300,38 +1265,46 @@ class TestFileManager : XCTestCase {
         // are throwable, an NSError is thrown instead which is more useful.
         let fm = FileManager.default
 
-        XCTAssertEqual(fm.homeDirectory(forUser: ""), URL(filePath: "/var/empty", directoryHint: .isDirectory))
-        XCTAssertEqual(NSHomeDirectoryForUser(""), "/var/empty")
+        #if os(Windows)
+        let defaultHomeDirectory = ProcessInfo.processInfo.environment["ALLUSERSPROFILE"]!
+        let emptyFileNameError: CocoaError.Code? = .fileReadInvalidFileName
+        #else
+        let defaultHomeDirectory = "/var/empty"
+        let emptyFileNameError: CocoaError.Code? = nil
+        #endif
+
+        XCTAssertEqual(fm.homeDirectory(forUser: ""), URL(filePath: defaultHomeDirectory, directoryHint: .isDirectory))
+        XCTAssertEqual(NSHomeDirectoryForUser(""), defaultHomeDirectory)
 
         XCTAssertThrowsError(try fm.contentsOfDirectory(atPath: "")) {
             let code = ($0 as? CocoaError)?.code
-            XCTAssertEqual(code, .fileReadNoSuchFile)
+            XCTAssertEqual(code, emptyFileNameError ?? .fileReadNoSuchFile)
```

Comment:
I'm open to suggestions on this. Previously in SCL-F, for empty paths we always threw `.fileReadInvalidFileName`. As part of the re-core, the Linux implementation changed to match the Darwin implementation which threw errors like `.fileReadNoSuchFile`. However, the Windows implementation still throws `.fileReadInvalidFileName`. For now, this just updates the test to account for the behavior today, but if we want (potentially in a separate change if that's cleaner) we can update the Windows implementation to throw the same error codes

Reply:
Yes, please! We should match the Darwin error codes if possible. The Windows path did reference the Linux path heavily for error conditions and semantics originally and the two diverged :(.

Reply:
Filed https://github.com/apple/swift-foundation/issues/882 to track standardizing this as a followup (I think it'd be great to land these tests now to prevent any other behavioral regressions on Windows and I'll followup shortly with a change to swift-foundation to throw the right error code and copy these tests to that repo as well